mono/metadata/threads.c

```diff
@@ -5002,6 +5015,66 @@ mono_thread_is_foreign (MonoThread *thread)
 	return info->runtime_thread == FALSE;
 }
 
+#ifndef HOST_WIN32
+static void
+threads_native_thread_join_lock (gpointer key, gpointer value)
+{
+	pthread_t thread = (pthread_t)key;
+	if (thread != pthread_self ()) {
+		MONO_ENTER_GC_SAFE;
+		/* This shouldn't block */
+		mono_threads_join_lock ();
+		mono_native_thread_join (thread);
+		mono_threads_join_unlock ();
+		MONO_EXIT_GC_SAFE;
+	}
+}
+inline static void
+threads_native_thread_join_nolock (gpointer key, gpointer value)
```

Comment:
These are not perf critical, they shouldn't be inline.

Reply:
OK